File: src/main/java/net/siegemc/core/items/CustomWeapon.java
package net.siegemc.core.items;

import lombok.Getter;
import lombok.Setter;
import org.bukkit.inventory.ItemStack;

import java.util.HashMap;
import java.util.Map;

public class CustomWeapon extends CustomItem {
    @Getter @Setter private double damage = 0;
    
    public CustomWeapon(String itemName, ItemStack rawItem) {
        super(itemName, rawItem);
    }
    
    public CustomWeapon(Map<String, Object> data) {
        super(data);
        setDamage((Double) data.get("weapon.damage"));
    }
    
    @Override
    public HashMap<String,Object> serialize() {
        HashMap<String, Object> r = super.serialize();
        r.put("weapon.damage", getDamage());
        return r;
    }
}

File: src/main/kotlin/net/siegemc/core/items/types/equipment/weapons/CustomWeapon.kt
package net.siegemc.core.items.types.equipment.weapons

import net.siegemc.core.v2.enums.ItemTypes
import net.siegemc.core.items.types.equipment.CustomEquipment

abstract class CustomWeapon : CustomEquipment() {
    override val type: ItemTypes = ItemTypes.WEAPON

    //abstract val damage: Double



}